### PR TITLE
exercises- 05 feb 2024

### PR DESCRIPTION
**Pull request by sabrina-graur**
Create a Java program to manage a list of books. Each book has a title, author, genre, and a publication year.
Implement methods to filter and display books published after a certain year (e.g. 1950) using Java Streams.
Additionally, create a method to calculate the average publication year of all books in the list.
Tips:
1. Can use:
               @Override
    public String toString() {
        return title + " by " + author + " (" + genre + ", " + publicationYear + ")";
    }
2. Could use books.forEach(System.out::println); to display the values of the "books" object.
3. Cound use Arrays.asList() to add the books to the list.
4. You're methods "filterBooksByPublicationYear" and "calculateAveragePublicationYear" should use Java Streams.

### CODE REVIEW

File: src/main/java/mainclass/exercises/streams/FavoriteBook.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package mainclass.exercises.streams;
+
+public class FavoriteBook {
+    String title;
+    String author;
+    String genre;
+    int year;
```
Comment (tudor-chirosca) [inline src/main/java/mainclass/exercises/streams/FavoriteBook.java:7]: Use the private access modifier, please.

File: src/main/java/mainclass/exercises/streams/Library.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        books.forEach(System.out::println);
+        System.out.println("\nThe filtered one: ");
+        filterBooksByPublicationYear(books).stream().forEach(System.out::println);
+        System.out.println("\nThe average year of publication is: ");
+        calculateAveragePublicationYear(books).stream().forEach(System.out::println);
+    }
+
+    public static List<FavoriteBook> filterBooksByPublicationYear(List<FavoriteBook> book) {
+        return book.stream()
+                .filter(b -> b.year >= 1980).toList();
```
Comment (tudor-chirosca) [inline src/main/java/mainclass/exercises/streams/Library.java:23]: The task is to "display books published after a certain year (e.g. 1950)".

File: src/main/java/mainclass/exercises/streams/FavoriteBook.java (hunk 1 above)
Comment (sabrina-graur) [reply src/main/java/mainclass/exercises/streams/FavoriteBook.java]: updated, thank you!

File: src/main/java/mainclass/exercises/streams/Library.java (hunk 2 above)
Comment (sabrina-graur) [reply src/main/java/mainclass/exercises/streams/Library.java]: updated, thank you!

**Review by tudor-chirosca — APPROVED** [review]

**Pull request merged**